src/database/sql_database.cc

```diff
@@ -718,25 +718,25 @@ void SQLDatabase::addObject(const std::shared_ptr<CdsObject>& obj, int* changedC
 
 void SQLDatabase::updateObject(const std::shared_ptr<CdsObject>& obj, int* changedContainer)
 {
-    std::vector<std::shared_ptr<AddUpdateTable>> data;
+    std::vector<AddUpdateTable> data;
     if (obj->getID() == CDS_ID_FS_ROOT) {
         std::map<std::string, std::string> cdsObjectSql;
 
         cdsObjectSql["dc_title"] = quote(obj->getTitle());
         setFsRootName(obj->getTitle());
         cdsObjectSql["upnp_class"] = quote(obj->getClass());
 
-        data.push_back(std::make_shared<AddUpdateTable>(CDS_OBJECT_TABLE, std::move(cdsObjectSql), Operation::Update));
+        data.emplace_back(CDS_OBJECT_TABLE, std::move(cdsObjectSql), Operation::Update);
     } else {
         if (IS_FORBIDDEN_CDS_ID(obj->getID()))
             throw_std_runtime_error("Tried to update an object with a forbidden ID ({})", obj->getID());
         data = _addUpdateObject(obj, Operation::Update, changedContainer);
     }
 
     beginTransaction("updateObject");
-    for (auto&& addUpdateTable : data) {
+    for (const auto& addUpdateTable : data) {
```

Comment:
all for loops use auto&& in the codebase. I would not change that. auto&& here evaluates to const auto&.